appservice/src/tree/AppSettingsTreeItem.ts

```diff
@@ -10,12 +10,19 @@ import { AppSettingTreeItem } from './AppSettingTreeItem';
 import { getThemedIconPath } from './IconPath';
 import { ISiteTreeRoot } from './ISiteTreeRoot';
 
-export function validateAppSettingKey(settings: StringDictionary, newKey?: string, oldKey?: string): string | undefined {
-    newKey = newKey ? newKey.trim() : '';
-    oldKey = oldKey ? oldKey.trim().toLowerCase() : oldKey;
+export function validateAppSettingKey(settings: StringDictionary, newKey?: string, oldKey?: string, isLinux: boolean = false): string | undefined {
```

Comment:
As discussed offline, let's move this into AppSettingsTreeItem and have the extensions use the treeItem to validate.  We won't need the `isLinux` boolean anymore as it will have access to the root treeItem.